week-3/todo-list/script.js

```diff
@@ -1,25 +1,33 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
-}
-
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
-let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
-];
-
-populateTodoList(todos);
+let list = document.getElementById("todo-list");
+const addTodo = document.querySelector("#add");
+const todos = [];
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
-function addNewTodo(event) {
+function addNewTodo() {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  const input = document.querySelector('input');
+  if (input.value != "") {
+    const li = document.createElement("li");
+    list.appendChild(li);
+    li.innerHTML = input.value;
```

Comment:
Here you're setting `innerHTML` to some raw string that was entered by the user.

This can cause some security issues - have a read of [this page](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#security_considerations) for some information about this.

Instead, I'd recommend setting `li.innerText` or `li.textContent` instead of `li.innerHTML` here.

(This is only a concern if you can't trust the input you're using as the value, e.g. if it's supplied by the user - if you control what the input is, like your `htmlContent` variable below, this isn't a problem).

Reply:
I got it. I fixed it. thanks.